### ISSUE TITLE
FATAL: URI Syntax Exception for "<redirectUri>"

### ISSUE THREAD

**Issue by DaChunky** [maximilianvoss/spotify-toniebox-sync#5]
Hi, 

first of all I have to say, that I am exited to find such a project here. Thank you for your effort!

I clone the repo and build like suggested. Everything works fine, Afterwards I edited the "_remote.repositories" file. I added 
`spotify.clientId=lxvkslfdgko1289378 # my client id`
`spotify.clientSecret=skldjflsdkj28137891 # my secret`
`spotify.redirectUri=http://aisjd93042903.mytonie.de # my redirectUri`
The letters and numbers above are randomly. Originally I added by values from the developer page. Afterwards I run
`java -jar spotify-toniebox-sync-[VERSION]-jar-with-dependencies.jar --apicode`
And got:
`FATAL: : URI Syntax Exception for "<redirectUri>"`
`Exception in thread "main" java.lang.NullPointerException at com.wrapper.spotify.requests.authorization.authorization_code.AuthorizationCodeUriRequest$Builder.redirect_uri(AuthorizationCodeUriRequest.java:74)`
`        at com.wrapper.spotify.SpotifyApi.authorizationCodeUri(SpotifyApi.java:406) at rocks.voss.spotifytonieboxsync.SpotifyAuthenticationSetup.getSignInUrl(SpotifyAuthenticationSetup.java:29)`
`        at rocks.voss.spotifytonieboxsync.Application.main(Application.java:40)`
What I am doing wrong? Do I have to add my spotify credentials in another file?

Thanks for your help

**Comment by DaChunky** [comment]
Ok, now I am a step further. I found the "spotify-toniebox-sync.properties" file under "src/main/resources". After build again, I got a URI as described. I opened the URI in a browser, accept the condinitions and was redirected to my defined URI. Unfortunatly there is no data field within the URI:  `https://0a5c05dc1cdf4433b9d23567d99f0dfc.mytoine.de/?code=AQD_1X5hijywT1Uo6F6gJLmh87390N9D5hbLa6PpoVrvFhvMHSe7AKAzreVbUTLeGJiBfFM3FgY3y-y_ExkSB3csgcwfy5Qs-hzxIavP-h3rWL5wyHX9zCL0FrmFJwty1-QWr5HUcVEfxcb96Wd9mQhALuIoEuv-LKW6Gn3CBKPDqy1JhslKAkCoh3rPN1Ztfn5PBt3MJRUbXbWkRRy8rSZ6qBDZtIxwR9AykXlNruJc5Js5b3VnSvalvOqaYRqYDrNOdPmlepHKXlB4EvGivLguT8MAcuD6WldNO25iN4YTh11HRSGZ_O_KgOlzhyN8yGwIg-8Insi7p4dufU3r5QPj3avpeLorsTI1lZc&state=x4xkmn9pu3j6ukrs8n`
I could only find a "code" and a "state" field.

**Comment by udovic2100** [comment]
It says data parameter in the readme but you actually need the code without the state part. Make sure you don’t cope the = or the & sign. Let me know how it works for you. I’m stuck at recording some tracks.

**Comment by DaChunky** [comment]
Ok, thanks.
But after I call  
` java -jar spotify-toniebox-sync-1.3-SNAPSHOT-jar-with-dependencies.jar --code "AQD_1X5hij....`
I got no output. 

EDIT:
So, after doing the procedure from the scratch, I got a refresh token. I put it into the properties file and rebuild the project (have to do I this step every time I changed the properties file or read the app the properies file on each start?). Afterwards I call
`java -jar spotify-toniebox-sync-1.3-SNAPSHOT-jar-with-dependencies.jar --playlist "EnglFavs" --tonie "Kreativ-Tonie"`
where "EnglFavs" is on of my own Playlist and "Kreativ-Tonie" the name of the creative tonie of my child. Furthermore I create a cache path, where the files should be stored in. Unfortunately I get no items in the cache path although I put the tonie on the box and start the deamon as suggested. I got following error:
`com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "ownerName" (class rocks.voss.toniebox.beans.toniebox.Household), not marked as ignorable (6 known properties: , "access", "canLeave", "id", "image", "name", "foreignCreativeTonieContent"])`
 `at [Source: N/A; line: -1, column: -1] (through reference chain: rocks.voss.toniebox.beans.toniebox.Household["ownerName"])`
        `at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:79)`
        `at com.fasterxml.jackson.databind.DeserializationContext.reportUnknownProperty(DeserializationContext.java:555)`
        `at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:708)`
        `at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1160)`
        `at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:315)`
        `at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:121)`
        `at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:151)`
        `at com.fasterxml.jackson.databind.deser.std.ObjectArrayDeserializer.deserialize(ObjectArrayDeserializer.java:18)`
`        at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:2860)`
`        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:1569)`
`        at com.fasterxml.jackson.databind.ObjectMapper.treeToValue(ObjectMapper.java:1902)`
`        at rocks.voss.toniebox.beans.Transformer.createBean(Transformer.java:23)`
`        at rocks.voss.toniebox.beans.Transformer.createBean(Transformer.java:16)`
`        at rocks.voss.toniebox.RequestHandler.executeRequest(RequestHandler.java:149)`
`        at rocks.voss.toniebox.RequestHandler.executeGetRequest(RequestHandler.java:106)`
`        at rocks.voss.toniebox.RequestHandler.getHouseholds(RequestHandler.java:55)`
`        at rocks.voss.toniebox.TonieHandler.getHouseholds(TonieHandler.java:55)`
`        at rocks.voss.spotifytonieboxsync.Application.main(Application.java:65)`

Where should I add the "ownerName"? In which XML?
